src/java/org/apache/cassandra/io/sstable/format/SSTableFormat.java

```diff
@@ -156,23 +168,23 @@ public static class Types
         {
             // the base data for an sstable: the remaining components can be regenerated
             // based on the data component
-            public static final Component.Type DATA = Component.Type.createSingleton("DATA", "Data.db", null);
+            public static final Component.Type DATA = Component.Type.createSingleton("DATA", "Data.db", true, null);
             // file to hold information about uncompressed data length, chunk offsets etc.
-            public static final Component.Type COMPRESSION_INFO = Component.Type.createSingleton("COMPRESSION_INFO", "CompressionInfo.db", null);
+            public static final Component.Type COMPRESSION_INFO = Component.Type.createSingleton("COMPRESSION_INFO", "CompressionInfo.db", true, null);
             // statistical metadata about the content of the sstable
-            public static final Component.Type STATS = Component.Type.createSingleton("STATS", "Statistics.db", null);
+            public static final Component.Type STATS = Component.Type.createSingleton("STATS", "Statistics.db", true, null);
             // serialized bloom filter for the row keys in the sstable
-            public static final Component.Type FILTER = Component.Type.createSingleton("FILTER", "Filter.db", null);
+            public static final Component.Type FILTER = Component.Type.createSingleton("FILTER", "Filter.db", true, null);
             // holds CRC32 checksum of the data file
-            public static final Component.Type DIGEST = Component.Type.createSingleton("DIGEST", "Digest.crc32", null);
+            public static final Component.Type DIGEST = Component.Type.createSingleton("DIGEST", "Digest.crc32", true, null);
             // holds the CRC32 for chunks in an uncompressed file.
-            public static final Component.Type CRC = Component.Type.createSingleton("CRC", "CRC.db", null);
+            public static final Component.Type CRC = Component.Type.createSingleton("CRC", "CRC.db", true, null);
             // table of contents, stores the list of all components for the sstable
-            public static final Component.Type TOC = Component.Type.createSingleton("TOC", "TOC.txt", null);
+            public static final Component.Type TOC = Component.Type.createSingleton("TOC", "TOC.txt", false, null);
             // built-in secondary index (may exist multiple per sstable)
-            public static final Component.Type SECONDARY_INDEX = Component.Type.create("SECONDARY_INDEX", "SI_.*.db", null);
+            public static final Component.Type SECONDARY_INDEX = Component.Type.create("SECONDARY_INDEX", "SI_.*.db", true, null);
             // custom component, used by e.g. custom compaction strategy
-            public static final Component.Type CUSTOM = Component.Type.create("CUSTOM", null, null);
+            public static final Component.Type CUSTOM = Component.Type.create("CUSTOM", null, true, null);
```

Comment:
I guess not all CUSTOM components should be streamable?

Is the plan here to just create something like the other `Types` inner classes, but in SAI's `V1OnDiskFormat`? If we did that, we could specify "streamability" there in those types. For example:

`Component.Type TERMS_DATA = Component.Type.create("TERMS_DATA", "SAI\\+.*\\+TermsData.db", true, null);`

This is really where all the stuff in `IndexComponent` probably should have gone in the first place.

CC @mike-tr-adamson

Reply:
...or more completely, something like...

```
public static class Components
{
    public static class Types extends AbstractSSTableFormat.Components.Types
    {
        public static final Component.Type TERMS_DATA = Component.Type.create("TermsData", "SAI\\+.*\\+TermsData.db", true, null);
...
    }

    public final static Component TERMS_DATA = Types.TERMS_DATA.getSingleton();
...
}
```

Reply:
Yes, that is the plan. I haven't finished that yet.

Reply:
Fixed